spec/features/assignments/edit_spec.rb

```diff
@@ -39,4 +44,70 @@
                                     text: month_date.strftime('%-B %G')
     end
   end
+  context 'Viewing the page' do
+    it 'displays the correct owner' do
+      last_name = assignment.user.last_name
+      visit edit_roster_assignment_url(roster, assignment)
+      expect(page).to have_selector :select, text: last_name
+    end
+    it 'displays the start date' do
+      visit edit_roster_assignment_url(roster, assignment)
+      expect(find_field('Start date').value)
+        .to eq start_date.strftime('%Y-%m-%d')
+    end
+    it 'displays the end date' do
+      visit edit_roster_assignment_url(roster, assignment)
+      expect(find_field('End date').value)
+        .to eq end_date.strftime('%Y-%m-%d')
+    end
+  end
+  context 'changing the assignment' do
+    let(:last_name) { user.last_name }
+    it 'updates the assignment' do
+      # Visit the index on the correct month to ensure the form
+      # submit returns us to the correct month
+      visit roster_assignments_url(roster, date: date_today)
+      visit edit_roster_assignment_url(roster, assignment)
+      fill_in('End date', with: date_today)
+      click_button 'Save'
+      # Since the assignment is now 5 days long
+      expect(page).to have_link last_name,
+                                href: edit_roster_assignment_url(roster.id,
+                                                                 assignment
+                                                                     .id),
```

Comment:
🤢 

In this case

```ruby
expect(page).to have_link last_name,
  edit_roster_assignment_url(roster, assignment)
```

would be better.

Note that you don't have to explicitly pass the IDs of the two objects. The objects themselves will do.